net-tools/netwatch/src/netmon/linux.rs

```diff
@@ -77,67 +82,70 @@ impl RouteMonitor {
                                 if addrs.contains(addr) {
                                     // already cached
                                     continue;
                                 } else {
-                                    addrs.insert(addr.clone());
+                                    addrs.insert(*addr);
                                     sender.send(NetworkMessage::Change).await.ok();
                                 }
                             }
                         }
-                        RtnlMessage::DelAddress(msg) => {
+                        RouteNetlinkMessage::DelAddress(msg) => {
                             trace!("DELADDR: {:?}", msg);
                             let addrs = addr_cache.entry(msg.header.index).or_default();
-                            if let Some(addr) = get_nla!(msg, address::Nla::Address) {
+                            if let Some(addr) = get_nla!(msg, address::AddressAttribute::Address) {
                                 addrs.remove(addr);
                             }
                             sender.send(NetworkMessage::Change).await.ok();
                         }
-                        RtnlMessage::NewRoute(msg) | RtnlMessage::DelRoute(msg) => {
+                        RouteNetlinkMessage::NewRoute(msg) | RouteNetlinkMessage::DelRoute(msg) => {
                             trace!("ROUTE:: {:?}", msg);
 
                             // Ignore the following messages
-                            let table = get_nla!(msg, route::Nla::Table)
+                            let table = get_nla!(msg, route::RouteAttribute::Table)
                                 .copied()
                                 .unwrap_or_default();
-                            if let Some(dst) = get_nla!(msg, route::Nla::Destination) {
-                                let dst_addr = match dst.len() {
-                                    4 => TryInto::<[u8; 4]>::try_into(&dst[..])
-                                        .ok()
-                                        .map(IpAddr::from),
-                                    16 => TryInto::<[u8; 16]>::try_into(&dst[..])
-                                        .ok()
-                                        .map(IpAddr::from),
-                                    _ => None,
-                                };
-                                if let Some(dst_addr) = dst_addr {
-                                    if (table == 255 || table == 254)
-                                        && (dst_addr.is_multicast() || is_link_local(dst_addr))
-                                    {
-                                        continue;
+                            if let Some(dst) = get_nla!(msg, route::RouteAttribute::Destination) {
+                                match dst {
+                                    route::RouteAddress::Inet(addr) => {
+                                        if (table == 255 || table == 254)
```

Comment:
I think I did this right but would like some extra 👀  on it @dignifiedquire

Reply:
yeah seems correct